[WIP at best] Quick and dirty tcp support

klausenbusk opened the pull request:
Featuring:
* Improper error management
* Impropper shutdown/reload support
* Bad code
* Breaking code
* Ineffective code
* A lot of "funny" prints/noise.

Config like:
```
[entryPoints]
  [entryPoints.tcp]
  mode = "tcp"
  address = ":2222"

[file]
[backends]
  [backends.backend1]
    [backends.backend1.servers.server1]
    url = "192.168.10.50:22"

[frontends]
  [frontends.frontend1]
  backend = "backend1"
  entrypoints = ["tcp"]
    [frontends.frontend2.routes.test_1]
    rule = "Host:backend2.com"
```

dtomcej commented:
This sort of features has been requested by multiple users.

We have been discussing cmux as a possible solution. Have you looked at this?

klausenbusk commented:
> We have been discussing cmux as a possible solution. Have you looked at this?

Nope, I haven't, but it could definitely make it simpler. I will take a closer look, if I can find some time.

How do you think the configuration should look like? A `tcp`/mode switch for both frontend/backend?

andrewrynhard commented:
@dtomcej Are your discussions in a public forum? I have a need to use TCP with Kubernetes and would like to get involved if I could.

dtomcej commented:
@andrewrynhard you can chat with us on slack!

http://traefik.slack.com/

klausenbusk commented:
I'm closing this as we have switched to Kubernetes and therefor switched to the [Nginx ingress controller](https://github.com/kubernetes/ingress/tree/master/controllers/nginx). Also https://github.com/containous/traefik/issues/1174 is a very serious issue, which is now fixed.
